### ISSUE TITLE
TFLiteConverter drops a leading broadcast dimension after MatMul followed by Add

### ISSUE THREAD

**Issue by laolvfan** [tensorflow/tensorflow#124048]
### 1. System information

- OS: Linux x86_64
- TensorFlow builds tested:
  - `2.20.0-dev0+selfbuilt`
  - `2.22.0-dev20260531` (`tf-nightly`)
- Python: 3.9 and 3.11 respectively
- CPU execution

### 2. Minimal reproducer

```python
import tensorflow as tf


class Model(tf.keras.Model):
    def __init__(self):
        super().__init__()
        self.weight = tf.Variable([[1.0, 2.0], [3.0, 4.0]])
        self.bias = tf.Variable([[5.0], [6.0]])

    def call(self, x):
        x = tf.reshape(x, [3, 2])
        x = tf.linalg.matmul(x, self.weight)
        return x + tf.reshape(self.bias, [1, 1, 2])


model = Model()
x = tf.constant([[[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]]])

expected = model(x).numpy()

converter = tf.lite.TFLiteConverter.from_keras_model(model)
interpreter = tf.lite.Interpreter(model_content=converter.convert())
key, signature = next(iter(interpreter.get_signature_list().items()))
actual = next(iter(interpreter.get_signature_runner(key)(
    **{signature["inputs"][0]: x.numpy()}
).values()))

print("eager :", expected.shape, expected)
print("tflite:", actual.shape, actual)
```

### 3. Observed result

Both tested builds print:

```text
eager : (1, 3, 2) [[[12. 16.]
                    [20. 28.]
                    [28. 40.]]]
tflite: (3, 2) [[12. 16.]
                 [20. 28.]
                 [28. 40.]]
```

The element values are preserved, but the converted model returns rank 2 instead of rank 3 and drops the leading dimension of size 1. This changes the TensorFlow program's observable output shape.

The same result occurs with raw `set_tensor(...); invoke()` instead of the signature runner.

### 4. Expected result

TFLite should preserve TensorFlow broadcasting semantics and return shape `(1, 3, 2)` with the same values as eager execution.

### 5. Additional narrowing

- Removing the final addition makes both eager and TFLite return `(3, 2)`.
- Keeping the addition but removing `relu` still reproduces the rank loss.
- The issue therefore does not require a nonlinear activation or random inputs.

**Comment by laolvfan** [comment]
I re-tested the minimal reproducer from this issue with the newest `tf-nightly` available in my environment: `2.22.0-dev20260726` on Linux x86_64, CPU, Python 3.11.

The issue no longer reproduces. The eager and TFLite outputs now both have shape `(1, 3, 2)`, and their values match exactly. I repeated the check in three fresh Python processes; all three runs passed.
